### DISCUSSION TITLE
It would be nice if the navigation icon settings could be controlled by config.

### DISCUSSION THREAD

**Discussion by dninomiya** [facebook/docusaurus#8103] [Ideas]
<img width="328" alt="image" src="https://user-images.githubusercontent.com/5842851/190587709-564ef90c-2703-46de-b5c3-82c5faa3ab18.png">

Many sites in the [showcase](https://docusaurus.io/showcase) have iconized GitHub and Discord. This means that it is a feature that many users want. But currently we have to do a bit of hacky work in custom.css to make that happen.

I thought it would be useful to have an approach where setting `icon:true` etc. in the config automatically gives an icon according to the URL, or where the user can specify the path to the image file.

```js
// example

 {
  href: 'https://github.com/facebook/docusaurus',
  icon: true, // auto iconized
  position: 'right',
},

or

 {
  href: 'https://github.com/facebook/docusaurus',
  imageURL: '/images/xxxx',
  label: 'GitHub', // as area-label
  position: 'right',
},
```

Icons for major services such as GitHub and Discord are most often requested. Therefore, the user should avoid the trouble of obtaining those logos independently and setting the image path.

I would prefer an option to auto-iconify based on the URL, preferably using something like [Simpe Icons](https://simpleicons.org/?q=github).

I love that this product can be used immediately with no setup or customization required. Please consider it.

Thank you for the wonderful framework!

**Comment by slorber** [comment]
You can also create custom navbar items. For now it's a workaround (https://github.com/facebook/docusaurus/issues/7227) but we plan to support this first-class in the future.

Now I agree that using a predefined set of common dev-related social icons could be useful and save time to a lot of users 👍

**Reply by dninomiya** [reply]
Nice to hear about that plan! thanks for the great activity
